internal/controller/pipelinerun_controller.go

```diff
@@ -95,15 +102,52 @@ func (r *PipelineRunReconciler) launchUpToNPipelineRuns(numToLaunch int, existin
 	for _, pipelineRun := range existingPipelineRuns.Items {
 		if pipelineRun.IsPending() {
 			original := pipelineRun.DeepCopy()
+
+			// renew github secret if it is no more valid or too old
+			if pipelineRun.Labels[MintMakerAppstudioLabel] == "github" && time.Since(pipelineRun.ObjectMeta.CreationTimestamp.Time) > GhTokenUsageWindow {
+				var component appstudiov1alpha1.Component
+				var ghComponent *github.Component
+				componentName := pipelineRun.Labels["mintmaker.appstudio.redhat.com/component"]
+				componentKey := types.NamespacedName{Namespace: "mintmaker", Name: componentName}
+				err = r.Client.Get(ctx, componentKey, &component)
+				if err != nil {
+					return err
+				}
+				timestamp, err := strconv.ParseInt(component.Labels["mintmaker.appstudio.redhat.com/reconcile-timestamp"], 10, 64)
+				if err != nil {
+					return err
+				}
+				ghComponent, err = github.NewComponent(&component, timestamp, r.Client, ctx)
+				if err != nil {
+					return err
+				}
+				token, err := (*ghComponent).GetToken() // renews token
+				if err != nil {
+					return err
+				}
+				appSecret := corev1.Secret{}
+				appSecretKey := types.NamespacedName{Namespace: "mintmaker", Name: pipelineRun.Name}
+				err = r.Client.Get(ctx, appSecretKey, &appSecret)
+				if err != nil {
+					return err
+				}
+				tokenInfo := github.TokenInfo{
+					Token: token, ExpiresAt: time.Now().Add(GhTokenValidity),
+				}
+				appSecret.Data["renovate-token"] = []byte(base64.StdEncoding.EncodeToString([]byte(tokenInfo.Token)))
```

Comment:
I don't see where the secret is being updated. You need to call update/patch to make the change.